lib/machinery_helper.rb

```diff
@@ -52,15 +57,19 @@ def inject_helper
   end
 
   def run_helper(scope)
-    json = @system.run_command(
-      File.join(
-        Machinery::REMOTE_HELPERS_PATH, "machinery-helper"
-      ), stdout: :capture, stderr: STDERR
-    )
+    json = @system.run_command(@remote_helpers_path, stdout: :capture, stderr: STDERR)
     scope.set_attributes(JSON.parse(json))
   end
 
   def remove_helper
-    @system.remove_file(File.join(Machinery::REMOTE_HELPERS_PATH, "machinery-helper"))
+    @system.remove_file(@remote_helpers_path)
+  end
+
+  def is_supported?
```

Comment:
`is_supported?` is a too generic name. The method name should make it clear what it does, especially when thinking from the point of view of the calling code.